tests/test_utils.py

```diff
@@ -390,3 +390,8 @@ def test_format_string_values(self):
         self.assertEqual(fmt["Key3"], "aa")
         self.assertEqual(fmt["Key4"][2], "aa")
         self.assertEqual(fmt["Key4"][1]["K"], "bb")
+
+        self.assertEqual(
```

Comment:
shouldn't this test be asserting something like this:

```python
         self.assertEqual(
             utils.format_string_values(
                 "{1}{account}", account='1234'),
             "{1}1234")
```

Reply:
no, that would fail, we get fallbacks (aka return original string) for regex patterns, not partial evaluation.

Reply:
Oh, ok. I thought the original intent of the fix was to return partially evaluated strings. If there are docs for var interpolation we should include what types of regex expressions are legal/illegal

Reply:
it is partial against a dictionary, its not partial against an individual string value in the dictionary. with this all regex strings are valid, just not interpolated if they include `{}` usage.